sessionctx/stmtctx/stmtctx.go

```diff
@@ -611,13 +612,34 @@ func (sc *StatementContext) SetPlanHint(hint string) {
 	sc.planHint = hint
 }
 
+// PlanCacheType is the flag of plan cache
+type PlanCacheType int
+
+const (
+	// DefaultNoCache no cache
+	DefaultNoCache PlanCacheType = iota
+	// SessionPrepared session prepared plan cache
+	SessionPrepared
+	// SessionNonPrepared session non-prepared plan cache
+	SessionNonPrepared
+)
+
 // SetSkipPlanCache sets to skip the plan cache and records the reason.
-func (sc *StatementContext) SetSkipPlanCache(reason error) {
+func (sc *StatementContext) SetSkipPlanCache(cacheType PlanCacheType, reason error) {
```

Comment:
It seems like we don't need to add `cacheType` as an argument here since we can just get it from `sc.CacheType`.

Reply:
Absolutely right